src/main/java/hyperledger/besu/java/rest/client/dto/TopicFilter.java

```diff
@@ -0,0 +1,38 @@
+package hyperledger.besu.java.rest.client.dto;
+
+import hyperledger.besu.java.rest.client.config.EthEventsProperties;
+import java.util.List;
+import lombok.Builder;
+import lombok.Data;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.web3j.protocol.core.DefaultBlockParameter;
+import org.web3j.protocol.core.methods.request.EthFilter;
+
+@Slf4j
+@Data
+@Builder
+public class TopicFilter {
+  private DefaultBlockParameter fromBlock; // optional, params - defaults to latest for both
+  private DefaultBlockParameter toBlock;
+  private String blockHash; // optional, cannot be used together with fromBlock/toBlock
+  private List<String> address; // spec. implies this can be single address as string or list
+  private EthFilter ethFilter;
+
+  @Autowired EthEventsProperties ethEventsProperties;
+
+  // set these values from configuration
+
+  private void setEthFilterWithFromAndToBlock() {
+
+    ethFilter = new EthFilter(fromBlock, toBlock, address);
+  }
+
+  private void setEthFilterWithBlockHash() {
+    ethFilter = new EthFilter(blockHash);
+  }
+
+  private void setEthFilterWithBlockHashAndSingleSmartContract(String smartContractAddress) {
+    ethFilter = new EthFilter(blockHash, smartContractAddress);
+  }
```

Comment:
Are we using it somewhere ? In our case, the application teams won't be passing the start and end block number.

Reply:
corrected